contrib/src/main/java/org/apache/apex/malhar/contrib/kudu/AbstractKuduOutputOperator.java

```diff
@@ -0,0 +1,666 @@
+    }
+    opsErrorsInThisWindow = kuduClientHandle.getStatistics().getClientStatistic(Statistics.Statistic.OPS_ERRORS) -
+      totalOpsErrors;
+    bytesWrittenInThisWindow = kuduClientHandle.getStatistics().getClientStatistic(Statistics.Statistic.BYTES_WRITTEN) -
+      totalBytesWritten;
+    rpcErrorsInThisWindow = kuduClientHandle.getStatistics().getClientStatistic(Statistics.Statistic.RPC_ERRORS) -
+      totalRpcErrors;
+    writeOpsInThisWindow = kuduClientHandle.getStatistics().getClientStatistic(Statistics.Statistic.WRITE_OPS) -
+      totalWriteOps;
+    writeRPCsInThisWindow = kuduClientHandle.getStatistics().getClientStatistic(Statistics.Statistic.WRITE_RPCS) -
+      totalWriteOps;
+    totalOpsErrors = kuduClientHandle.getStatistics().getClientStatistic(Statistics.Statistic.OPS_ERRORS);
+    totalBytesWritten = kuduClientHandle.getStatistics().getClientStatistic(Statistics.Statistic.BYTES_WRITTEN);
+    totalRpcErrors = kuduClientHandle.getStatistics().getClientStatistic(Statistics.Statistic.RPC_ERRORS);
+    totalWriteOps = kuduClientHandle.getStatistics().getClientStatistic(Statistics.Statistic.WRITE_OPS);
+    totalWriteRPCs = kuduClientHandle.getStatistics().getClientStatistic(Statistics.Statistic.WRITE_RPCS);
+  }
+
+  private void buildGettersForPojoPayload()
+  {
+    Class payloadClass = getTuplePayloadClass();
+    checkNotNull(payloadClass,"Payload class cannot be null");
+    Field[] classFields = payloadClass.getDeclaredFields();
+    Schema schemaInfo = kuduTable.getSchema();
+    List<ColumnSchema> allColumns = schemaInfo.getColumns();
+    Set<String> allKuduTableColumnNames = new HashSet<>();
+    Map<String,ColumnSchema> normalizedColumns = new HashMap();
+    for ( ColumnSchema aColumnDef : allColumns) {
+      allColumnDefs.put(aColumnDef.getName(), aColumnDef);
+      normalizedColumns.put(aColumnDef.getName().toLowerCase(), aColumnDef);
+      allKuduTableColumnNames.add(aColumnDef.getName().toLowerCase());
+    }
+    List<ColumnSchema> primaryKeyColumns = schemaInfo.getPrimaryKeyColumns();
+    for (ColumnSchema primaryKeyInfo : primaryKeyColumns) {
+      primaryKeyColumnNames.add(primaryKeyInfo.getName());
+    }
+    Map<String,String> columnNameOverrides = getOverridingColumnNameMap();
+    if (columnNameOverrides == null) {
+      columnNameOverrides = new HashMap(); // to avoid null checks further down the line
+    }
+    for ( Field aFieldDef : classFields) {
+      String currentFieldName = aFieldDef.getName().toLowerCase();
+      if (allKuduTableColumnNames.contains(currentFieldName)) {
+        extractGetterForColumn(normalizedColumns.get(currentFieldName),aFieldDef);
+      } else {
+        if (columnNameOverrides.containsKey(aFieldDef.getName())) {
+          extractGetterForColumn(normalizedColumns.get(columnNameOverrides.get(aFieldDef.getName()).toLowerCase()),
+              aFieldDef);
+        }
+        if (columnNameOverrides.containsKey(aFieldDef.getName().toLowerCase())) {
```

Comment:
else if ??
If you have aFieldDef.getName() == aFieldDef.getName().toLowerCase() then getter will be created twice.